pkg/volume/csi/csi_attacher.go

```diff
@@ -630,3 +650,11 @@ func getDriverAndVolNameFromDeviceMountPath(k8s kubernetes.Interface, deviceMoun
 
 	return csiSource.Driver, csiSource.VolumeHandle, nil
 }
+
+// getNameForVolSource builds a unique name with sha256(specName,csiDriverName,NodeName).
+// This generates a predictable string used in building devicepath when an ephemeral volume
+// is encountered.
+func getNameForVolSource(specName, csiDriverName, nodeName string) string {
```

Comment:
I still am not sure how this is unique?

You could have two different pods with the same volume name scheduled to the same node.

Reply:
its unique within a pod, but not between pods. I think the comment above on how its used says thats ok for now, since it isnt actually used?

Reply:
This is used to generate the device mount path, which is then [globally marked as mounted](https://github.com/kubernetes/kubernetes/blob/155688b2f3521ffa913766086ab436eaae81639b/pkg/volume/util/operationexecutor/operation_generator.go#L628).  It might be that for ephemeral volumes, since device mount path is essentially ignored, then maybe that's not a problem.  Maybe it would be better to return an empty string rather than fake some path that isn't used.

Reply:
Oh sorry, I was confusing unique volume name with the device mount path.  We need to set some volume name.  It may conflict with other pods using the same csi driver, and maybe that's ok because we don't do anything in MountDevice for ephemeral volumes anyway.

Reply:
Wait sorry I'm confusing myself too much :(  This is used to generate the device path.

Reply:
@msau42 So, do you think this needs to change or is it ok to leave as is? 
I'd reiterate that under normal circumstances, ephemeral volumes for CSI should not participate in any storage operations other than mount/unmount.  This is there to satisfy Volume API since  the volume manager is not skipping MountDevice during attachment phase (even when plugin.CanSkip returns true)

Reply:
If we really are confident this isn't used anywhere, then I would rather have some really obvious fake name instead of trying to be unique but not actually.  So maybe just having the csiDriverName as the path is sufficient.